cmake/ProcessOptions.cmake

```diff
@@ -392,14 +392,17 @@ function( NEST_PROCESS_WITH_PYTHON )
   elseif ( ${with-python} STREQUAL "ON" )
 
     # Localize the Python interpreter and ABI
-    if ( ${CMAKE_VERSION} VERSION_LESS "3.18.0")
+    find_package( Python 3.8 QUIET COMPONENTS Interpreter Development.Module )
+    if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
-      message( WARNING "CMake 3.18+ is recommended for more universal Python bindings. If you encounter missing Python header or library file errors, try upgrading CMake.")
+      string( CONCAT PYABI_WARN "Could not locate Python ABI"
+        ", using shared libraries and header file instead."
+        " Please clear your CMake cache and build folder and verify that CMake"
+        " is up-to-date (3.18+)."
+      )
+      message( WARNING "${PYABI_WARN}")
     else()
       find_package( Python 3.8 REQUIRED Interpreter Development.Module )
-    endif()
-
-    if ( Python_FOUND )
```

Comment:
I think we need to re-add these to keep the logic as 
1. check if we can find any of Development or Development.Module
2. check if we found one of them and do stuff

Reply:
Yea, I noticed after checking the diff, hard to see the impact of the suggestions on GitHub sometimes :) fixed in 15b4614